aspnetcore-webapi/DotNetURLShortener/DotNetURLShortener/Services/UrlShortenerService.cs

```diff
@@ -0,0 +1,55 @@
+﻿using DotNetURLShortener.Contracts;
+
+namespace DotNetURLShortener.Services;
+
+public class UrlShortenerService : IUrlShortenerService
+{
+    private const string AlphaNumericChars = "0123456789abcdefghijklmnopqrstuvwxyzABCDEFGHIJKLMNOPQRSTUVWXYZ";
+    private const int ShortCodeLength = 5;
+
+    public Dictionary<string, string> UrlDict { get; set; } = [];
+
+    public string GetShortCode(string longUrl)
+    {
+        static string GenerateShortCode()
+        {
+            var shortCode = new char[ShortCodeLength];
+            for (var i = 0; i < ShortCodeLength; i++)
+            {
+                shortCode[i] = AlphaNumericChars[Random.Shared.Next(AlphaNumericChars.Length)];
+            };
+
+            return new string(shortCode);
+        }
+
+        foreach (var pair in UrlDict)
+        {
+            if (pair.Value == longUrl)
+            {
+                return pair.Key;
+            }
+        }
```

Comment:
Why not just use [Dictionary.TryGetValue](https://learn.microsoft.com/en-us/dotnet/api/system.collections.generic.dictionary-2.trygetvalue?view=net-8.0) here? We have a dictionary which as O(1) lookup, so we should take advantage of that.

Reply:
Hi @jshergal. Here, we are trying to retrieve the key(the shortcode) with a known value(a long URL). We use the TryGetValue() method in the second method in this class where we have a shortcode and wish to retrieve its long URL.

Reply:
@cjaynjoku - You are correct. That is my bad, yeah since you are looking for the value the best is a linear search. From a performance perspective, it might be worthwhile to maintain two dictionaries, but I suppose in general, if we are creating a URL shortener, we wouldn't be using dictionaries but an actual database and so we would have an index on the long URL value.

Reply:
That's right @jshergal, using an actual database would be better. However, the instruction in the issue's description was to avoid using databases and focus primarily on shortening the URL, probably to make the article easier to follow. I'll add this database caveat to the article.